### PR TITLE
Implements `TryFrom<Node> for f32`

### PR DESCRIPTION
**Pull request by odashi**
This pr adds `TryFrom<Node> for f32` which obtains scalar results by implicitly invoking the calculation.

Blocking change: #20

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -134,6 +125,18 @@ impl<'hw: 'op, 'op: 'g, 'g> PartialEq for Node<'hw, 'op, 'g> {
...
 
+/// Directly obtaining a scalar value from a node.
+impl<'hw: 'op, 'op: 'g, 'g> From<Node<'hw, 'op, 'g>> for f32 {
+    fn from(node: Node<'hw, 'op, 'g>) -> Self {
+        node.graph
+            .borrow_mut()
+            .calculate(node.step_id)
+            .unwrap()
+            .get_scalar_f32()
+            .unwrap()
```
Comment (tetsuok) [inline src/node.rs]: we can use `Node::calculate()`:

```suggestion
        node.calculate().unwrap().get_scalar_f32().unwrap()
```

**Review by tetsuok — APPROVED** [review]

File: src/node.rs [hunk 2]
```diff
@@ -134,6 +125,18 @@ impl<'hw: 'op, 'op: 'g, 'g> PartialEq for Node<'hw, 'op, 'g> {
...
+    }
```
Comment (vbkaisetsu) [inline src/node.rs]: from() must not fail, but I think this function will fail in some situations.
Could you use TryFrom instead?
https://doc.rust-lang.org/std/convert/trait.From.html

**Review by vbkaisetsu — CHANGES_REQUESTED** [review]

File: src/node.rs (hunk 2 above)
Comment (odashi) [reply src/node.rs]: Implementing both would be better. "must not fail" in this context means users understand that avoiding every error is users' responsibility, and it panics if some erroneous thing happens. Sometimes users know that the `Node` represents just a scalar and using `From` would be reasonable over `TryFrom` in such situation.

Comment (odashi) [reply src/node.rs]: As for the `i32`'s interface, it avoids `From` from larger types. I think avoiding `From` here makes sense.

Comment (odashi) [reply src/node.rs]: Fixed. 4b0b906

**Review by vbkaisetsu — APPROVED** [review]

**Pull request merged**